HorizonCore/src/main/java/com/hubspot/horizon/HttpConfig.java

```diff
@@ -63,10 +63,37 @@ public static Builder newBuilder() {
     return new Builder();
   }
 
+  /**
+   * The HttpClient will use this setting to throttle requests.
+   *
+   * If the client gets a request, but the max connections are already in use,
+   * the client will block indefinitely until it has a free connection for the request,
+   * then make the request once it's able.
+   *
+   * The default value is 100.
+   */
   public int getMaxConnections() {
     return maxConnections;
   }
 
+  /**
+   * Unlike {@link #getMaxConnections()}, the HttpClient will use this per-host setting to IMMEDIATELY FAIL any requests
+   * that would cause it to exceed the connections per host limit.
+   * The client does NOT block / wait for any amount of time if it gets a request when the max connections per host are already used up;
+   * instead, a {@link org.asynchttpclient.shaded.exception.TooManyConnectionsPerHostException} is thrown from the client immediately.
+   *
+   * Note, {@link #getConnectTimeoutMillis()} doesn't come into play in this connection limiting.
+   * Instead, the internal `acquireTimeout` is the timeout for acquiring a connection permit;
```

Comment:
Should this be connect timeout instead?

Reply:
No, I think what I meant to explain here is that the connect timeout can only come into play after the client has acquired a connection permit, so it's the acquireTimeout defaulting to 0 that causes "The client does NOT block / wait for any amount of time if it gets a request when the max connections per host are already used up" - and since acquireTimeout isn't exposed there's no way to have it not throw immediately when maxConnectionsPerHost is hit. The connect timeout should be for establishing a connection to the API after a connection permit is acquired (technically it could influence connection limiting if it was set way too high, but it isn't part of the immediate failure behavior).
Does that make sense? I could try to reword it if that might help